core/dbt/task/run_operation.py

```diff
@@ -66,8 +67,17 @@ def run(self) -> RunResultsArtifact:
         end = datetime.utcnow()
 
         package_name, macro_name = self._get_macro_parts()
-        fqn = [NodeType.Operation, package_name, macro_name]
-        unique_id = ".".join(fqn)
+        macro = self.manifest.find_macro_by_name(  # type: ignore[union-attr]
+            macro_name, self.config.project_name, package_name
+        )
+
+        try:
+            unique_id = macro.unique_id  # type: ignore[union-attr]
+            fqn = unique_id.split(".")
+        except AttributeError:
+            raise DbtRuntimeError(
```

Comment:
for my own understanding: if the macro is not found - we'll actually hit this error earlier on from the `self._run_unsafe` call via `adapter.execute_macro`, right? If so, maybe this should actually be a DbtInternalError - since we don't expect it to be reached if everything is running as expected.